### DISCUSSION TITLE
Change the validationSchema based on form value

### DISCUSSION THREAD

**Discussion by JoeVanGundy** [react-hook-form/react-hook-form#8190] [Q&A]
How can I change the schema for the form dynamically based off of the value of the form?
In this example, based on the value of a `<select>`, I need to swap out the validationSchema of the form (some fields are required and some aren't).

This is basically what I want to do, but I can't call `useWatch` before `useForm` so to my understanding, there is no way to access a form value in the schemaValidation?

```
const fieldValue = useWatch({ name: fieldValue });
```

```
const validationSchema = useMemo(() => {
   if(fieldValue === 'valueB') {
      return BranchBSchemaValidation;
   }
   return BranchASchemaValidation;
});
```

```
const methods = useForm({
    defaultValues,
    resolver: yupResolver(validationSchema),
});
```

```
useEffect(() => {
    reset(defaultValues);
}, [fieldValue]);
```

Thanks!!

**Comment by Moshyfawn** [comment]
Which validation library are you using?

**Reply by JoeVanGundy** [reply]
Hey @Moshyfawn , I'm using Yup

**Reply by bluebill1049** [reply]
could you use the `context` object at `useForm` and toggle the yup validation schema?`
